themes/qgrid/source/js/menu.js

```diff
@@ -21,17 +21,45 @@ function filter(e) {
 	const filter = input.value.toLowerCase();
 	const nav = document.getElementById('nav');
 	const li = nav.getElementsByTagName('li');
-
+	const h2 = document.getElementsByTagName('h2');
+	if (filter) location.hash = 'search=' + filter;
+	else location.hash = '';
 	for (let i = 0; i < li.length; i++) {
 		const a = li[i].getElementsByTagName('a')[0];
 		if (a) {
-			if (a.text.toLowerCase().indexOf(filter) > -1) {
+			const entr = a.text.toLowerCase().indexOf(filter);
+			if (entr > -1) {
+				const word = a.text.split('');
 				li[i].style.display = '';
+				for (let i = 0; i < filter.length; i++) {
+					word[entr + i] = '<font style=background-color:#efdf00>' + word[entr + i] + '</font>';
```

Comment:
won't it better to surround whole word with on tag? instead of wrapping each letter? also try to avoid implicit styles, use css classes please.